src/Kestrel.Core/Internal/Http2/Http2OutputProducer.cs

```diff
@@ -46,6 +46,8 @@ public class Http2OutputProducer : IHttpOutputProducer
             _dataWriteProcessingTask = ProcessDataWrites();
         }
 
+        public bool IsCompleted => _completed;
```

Comment:
It's not safe to check `_completed` outside the lock. I see that the way it's being used now is basically an optimization, but `HttpContext.Abort()` is not called frequently enough for this to be worth it. Remove this property.